packages/noya-state/src/reducers/application.ts

```diff
@@ -1257,6 +1277,231 @@ export function reducer(
         else draft.selectedThemeStyleGroup = '';
       });
     }
+    case 'setTextColor': {
+      const [, color] = action;
+
+      const pageIndex = getCurrentPageIndex(state);
+      const layerIndexPaths = getSelectedLayerIndexPaths(state);
+
+      return produce(state, (draft) => {
+        accessPageLayers(draft, pageIndex, layerIndexPaths).forEach((layer) => {
+          if (layer._class !== 'text') return;
+          const encoded = layer.style?.textStyle?.encodedAttributes;
+          const attributes = layer.attributedString.attributes[0].attributes;
```

Comment:
Would we want to do a `forEach` on `attributedString.attributes` to change all of them? (same for the other actions)

Reply:
I think I'll have to have add an index 🤔 to handle those attributes, because I am guessing the other attributes are for different areas of the text. I will add the foreach and later the conditionals for to access the right string